invokeai/app/util/profiler.py

```diff
@@ -0,0 +1,76 @@
+import cProfile
+from contextlib import suppress
+from logging import Logger
+from pathlib import Path
+from typing import Optional
+
+
+class Profiler:
+    """
+    Simple wrapper around cProfile.
+
+    Usage
+    ```
+      # Create a profiler
+      profiler = Profiler(logger, output_dir)
+      # Start a new profile
+      profiler.new("my_profile")
+      profiler.enable()
+      # Do stuff
+      profiler.disable()
+      profiler.dump()
+    ```
+
+    Visualize a profile as a flamegraph with [snakeviz](https://jiffyclub.github.io/snakeviz/)
+    ```sh
+      snakeviz my_profile.prof
+    ```
+
+    Visualize a profile as directed graph with [graphviz](https://graphviz.org/download/) & [gprof2dot](https://github.com/jrfonseca/gprof2dot)
+    ```sh
+      gprof2dot -f pstats my_profile.prof | dot -Tpng -o my_profile.png
+      # SVG or PDF may be nicer - you can search for function names
+      gprof2dot -f pstats my_profile.prof | dot -Tsvg -o my_profile.svg
+      gprof2dot -f pstats my_profile.prof | dot -Tpdf -o my_profile.pdf
+    ```
+    """
+
+    def __init__(self, logger: Logger, output_dir: Path, prefix: Optional[str] = None) -> None:
+        self.logger = logger
+        self.output_dir = output_dir
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+        self.profiler: Optional[cProfile.Profile] = None
+        self.profile_id: Optional[str] = None
+        self.prefix = prefix
```

Comment:
I think the intended public interface could be made more clear by marking private fields with a leading underscore. Something like:

```suggestion
        self._logger = logger
        self._output_dir = output_dir
        self._output_dir.mkdir(parents=True, exist_ok=True)
        self._profiler: Optional[cProfile.Profile] = None
        self._prefix = prefix
        
        self.profile_id: Optional[str] = None
```